R/TabPFN-fit.R

```diff
@@ -48,19 +48,142 @@
+#' account via the command line by running: hf auth login (Alternatively, you
+#' can set the HF_TOKEN environment variable with a read token)."
+#'
+#' The license contains provisions for "Non-Commercial Use Only" usage if that
+#' is relevant for you.
+#'
+#' To get an API key, use the `huggingface` link above, create an account, and
+#' then get an API key. Once you have that, put it in your `.Renviron` file in
+#' the form of:
+#'
+#' \preformatted{
+#' HF_TOKEN=your_api_key_value
+#' }
+#'
+#' The \pkg{usethis} function `edit_r_environ()` can be very helpful here.
+#'
+#' ## Python Installation
+#'
+#' You will need a working Python virtual environment with the correct packages
+#' to use these modeling functions.
+#'
+#' There are at least two ways to proceed.
+#'
+#' ### Ephemeral `uv` Install
+#'
+#' The first approach, which we *strongly suggest*, is to simply load this
+#' package and attempt to run a model. This will prompt \pkg{reticulate} to
+#' create an ephemeral environment and automatically install the required
+#' packages. That process would look like this:
+#'
+#' \preformatted{
+#'   > library(TabPFN)
+#'   >
+#'   > predictors <- mtcars[, -1]
+#'   > outcome <- mtcars[, 1]
+#'   >
+#'   > # XY interface
+#'   > mod <- tab_pfn(predictors, outcome)
+#'   Downloading uv...Done!
+#'   Downloading cpython-3.12.12 (download) (15.9MiB)
+#'    Downloading cpython-3.12.12 (download)
+#'   Downloading setuptools (1.1MiB)
+#'   Downloading scikit-learn (8.2MiB)
+#'   Downloading numpy (4.9MiB)
+#'
+#'   <downloading and installing more packages>
+#'
+#'    Downloading llvmlite
+#'    Downloading torch
+#'   Installed 58 packages in 350ms
+#'   > mod
+#'   TabPFN Regression Model
+#'
+#'   Training set
+#'   ℹ 32 data points
+#'   ℹ 10 predictors
+#' }
+#'
+#' The location of the environment can be found at
+#' `tools::R_user_dir("reticulate", "cache")`.
+#'
+#' See the documentation for [reticulate::py_require()] to learn more about this
+#' method.
+#'
+#' ### Manually created `venv` Virtual Environment
+#'
+#' Alternatively, you can use the functions in the \pkg{reticulate} package to
+#' create a virtual environment and install the required Python packages there.
+#' An example pattern is:
+#'
+#' \preformatted{
+#'   library(reticulate)
+#'   env_name <- "r-tabpfn"      # exact name can be different
+#'   py_version <- "3.13:latest" # or your version of choice
+#'
+#'   virtualenv_list()
+#'   py_inst <- install_python(py_version)
+#'
+#'   # Save py_inst value
+#'   <restart R>
+#'
+#'   library(reticulate)
+#'   use_python_version(py_version)
+#'   virtualenv_create(env_name, python = py_inst)
+#' }
+#'
+#' Once you have that virtual environment installed, you can load it _before_
+#' starting \pkg{TabPFN}:
+#'
+#' \preformatted{
+#'   reticulate::use_virtualenv("r-tabpfn")
+#' }
+#'
+#' **Important Note** for this method,. Due to how Python uses the OpenMP
```

Comment:
The OpenMP issues don't seem to happen with installs handled by `py_require()`. Does that make sense?

Reply:
I think the issue was patched upstream in llvmlite (we also encountered the same bug in ragnar).

https://github.com/numba/llvmlite/issues/1290#issuecomment-3299007584
https://github.com/numba/numba/issues/10231